src/python/pants/backend/python/tasks/pex_build_util.py

```diff
@@ -158,3 +164,39 @@ def _resolve_multi(interpreter, requirements, platforms, find_links):
       allow_prereleases=python_setup.resolver_allow_prereleases)
 
   return distributions
+
+
+def inject_req_libs_provided_by_setup_file(build_graph, local_built_dists, synthetic_address, binary_tgt=None):
+  """Build a requirements library from a local wheel.
+
+  :param build_graph: The build graph needed for injecting synthetic targets.
+  :param local_built_dists: A list of paths to locally built wheels to package into
+  requirements libraries.
+  :param synthetic_address: A generative address for addressing synthetic targets.
+  :param binary_tgt: An optional parameter to be passed only when called by the `python_binary_create`
+  task. This is needed to ensure that only python_dist targets in a binary target's closure are included
+  in the binary for the case where a user specifies mulitple binary targets in a single invocation of
+  `./pants binary`.
+  :return: a :class: `PythonRequirementLibrary` containing a local wheel and its
+  transitive dependencies.
```

Comment:
afaict, it doesn't actually include transitive dependencies here - just the surface requirement that aligns to the built wheel.